File: 42746.cpp
// 가장 큰 수

#include <string>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

bool cmp(string a, string b) {
    return (a + b > b + a);
}

string solution(vector<int> numbers) {
    string answer = "";
    vector<string> sub;

    for (int i = 0; i < numbers.size(); i++) {
        sub.push_back(to_string(numbers[i]));
    }
    sort(sub.begin(), sub.end(), cmp);
    if (sub.at(0) == "0") {
        answer += "0";
        return answer;
    } else {
        for (int i = 0; i < sub.size(); i++) {
            answer += sub[i];
        }
    }

    return answer;
}

File: 42746.py
# 가장 큰 수
# 2021-09-21

def solution(numbers):
    numbers = list(map(str, numbers))
    tmp = sorted(numbers, key=lambda x: x*3, reverse=True)
    return str(int(''.join(tmp)))

print(solution([3, 30, 34, 5, 9]))